### DISCUSSION TITLE
Able to select data, but not update it...

### DISCUSSION THREAD

**Discussion by Etheonor** [supabase/supabase#721] [Questions] [answered]
Hello everyone!

I have a weird issue and I'm not sure what's wrong with my code. 

I created a public.users table that will automaticaly create a row with id/email and an empty "twitter" column when a new user is created with the auth system. 

I can easily retrieve the information from that table with a simple : 

`let { data: socials, error } = await supabase
      .from("users")
      .select("twitter");`

(I added a policy on this table : (uid() = id))

But if I try something really simple like that : 

 `const { data, error } = await supabase
      .from("users")
      .update({ "twitter": "my account" })`

It doesn't work! Am I missing something? I also tried with .eq("id", user.id), even though I already have a policy that filter the result, but without any success.

Is this an issue with the policy I added? 

Thank you!

**Comment by Etheonor** [comment, marked as answer]
Ok I just needed to post a thread to find the answer the minute after. 

I forgot policies are just for one action, I created one for "Select", but not for "Update"... Now it works :) 

Maybe an improvement -> We need to click on "Advanced" to create a policy that apply to all options. Maybe the "All" tab can be available all the time?

**Reply by steve-chavez** [reply]
The hiding of `ALL` is on purpose. The problem with `ALL` is that it might make you think that it's the only policy needed, but in most cases you'll need an additional `DELETE` policy. This is because the `USING` clause of `ALL` will filter the rows the user can see, but it will not filter the rows the user can DELETE(like only owners of posts can delete them).

Once you define an `ALL` policy, you have to add a DELETE policy with the RESTRICTIVE keyword for filtering the rows that can be DELETEd. We thought this was not clear/intuitive, so we try to encourage users to define a policy for each statement.

**Reply by Etheonor** [reply]
This make sense, thank you for your answer ^^
